### PR TITLE
Bug 1554734 - Also support context-menu arrow key navigation 

### PR DESCRIPTION
**Pull request by AllegroFox**
added ArrowUp and ArrowDown listeners to the onKeyDown event on context menus; handled focus with DOM targeting. May try to switch to React refs to make the function a little more resilient.

### CODE REVIEW

File: content-src/components/ContextMenu/ContextMenu.jsx [hunk 1]
```diff
@@ -84,7 +100,7 @@ export class ContextMenuItem extends React.PureComponent {
     const {option} = this.props;
     return (
       <li role="menuitem" className="context-menu-item" >
-        <button className={option.disabled ? "disabled" : ""} onClick={this.onClick} onKeyDown={this.onKeyDown} tabIndex="0" >
+        <button className={option.disabled ? "disabled" : ""} tabIndex="0" onClick={this.onClick} onKeyDown={this.onKeyDown}>
```
Comment (k88hudson) [inline content-src/components/ContextMenu/ContextMenu.jsx]: what's the reason for this change?

File: content-src/components/ContextMenu/ContextMenu.jsx [hunk 2]
```diff
@@ -62,6 +62,17 @@ export class ContextMenuItem extends React.PureComponent {
     this.props.option.onClick();
   }
 
+  // This selects the correct node based on the key pressed
+  focusSibling(target, key) {
+    const parent = target.parentNode;
+    const closestSiblingSelector = (key === "ArrowUp") ? "previousSibling" : "nextSibling";
+    if (parent[closestSiblingSelector].firstElementChild) {
```
Comment (k88hudson) [inline content-src/components/ContextMenu/ContextMenu.jsx]: I think you also need to return early if `parent[closestSiblingSelector]` is falsey for the first/last child case

File: content-src/components/ContextMenu/ContextMenu.jsx (hunk 1 above)
Comment (AllegroFox) [reply content-src/components/ContextMenu/ContextMenu.jsx]: I'm not sure, I may have removed the tabIndex to test something and then put it back :)

File: content-src/components/ContextMenu/ContextMenu.jsx (hunk 2 above)
Comment (AllegroFox) [reply content-src/components/ContextMenu/ContextMenu.jsx]: `if (!parent[closestSiblingSelector]) {
      return;
    } else if (parent[closestSiblingSelector].firstElementChild) {
      parent[closestSiblingSelector].firstElementChild.focus();
    } else {
      parent[closestSiblingSelector][closestSiblingSelector].firstElementChild.focus();
    }`

Does that look about right?

Comment (AllegroFox) [reply content-src/components/ContextMenu/ContextMenu.jsx]: The behaviour seems to be the same

Comment (k88hudson) [reply content-src/components/ContextMenu/ContextMenu.jsx]: yep that's good, I see an error being thrown otherwise
